app/tydom/TydomClient.py

```diff
@@ -52,6 +54,56 @@ def __init__(
             self.cmd_prefix = ""
             self.ping_timeout = None
 
+    @staticmethod
+    def getTydomCredentials(login: str, password: str, macaddress: str):
+        """get tydom credentials from Delta Dore"""
+        try:
+            response = requests.get("https://deltadoreadb2ciot.b2clogin.com/deltadoreadb2ciot.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=B2C_1_AccountProviderROPC_SignIn")
+        
+            json_response = response.json()
+            response.close()
+            signin_url = json_response["token_endpoint"]
+            logger.debug("signin_url : %s", signin_url)
+
+            body, ct_header = encode_multipart_formdata(
+                {
+                    "username": f"{login}",
+                    "password": f"{password}",
+                    "grant_type": "password",
+                    "client_id": "8782839f-3264-472a-ab87-4d4e23524da4",
+                    "scope": "openid profile offline_access https://deltadoreadb2ciot.onmicrosoft.com/iotapi/video_config https://deltadoreadb2ciot.onmicrosoft.com/iotapi/video_allowed https://deltadoreadb2ciot.onmicrosoft.com/iotapi/sites_management_allowed https://deltadoreadb2ciot.onmicrosoft.com/iotapi/sites_management_gateway_credentials https://deltadoreadb2ciot.onmicrosoft.com/iotapi/sites_management_camera_credentials https://deltadoreadb2ciot.onmicrosoft.com/iotapi/comptage_europe_collect_reader https://deltadoreadb2ciot.onmicrosoft.com/iotapi/comptage_europe_site_config_contributor https://deltadoreadb2ciot.onmicrosoft.com/iotapi/pilotage_allowed https://deltadoreadb2ciot.onmicrosoft.com/iotapi/consent_mgt_contributor https://deltadoreadb2ciot.onmicrosoft.com/iotapi/b2caccountprovider_manage_account https://deltadoreadb2ciot.onmicrosoft.com/iotapi/b2caccountprovider_allow_view_account https://deltadoreadb2ciot.onmicrosoft.com/iotapi/tydom_backend_allowed https://deltadoreadb2ciot.onmicrosoft.com/iotapi/websocket_remote_access https://deltadoreadb2ciot.onmicrosoft.com/iotapi/orkestrator_device https://deltadoreadb2ciot.onmicrosoft.com/iotapi/orkestrator_view https://deltadoreadb2ciot.onmicrosoft.com/iotapi/orkestrator_space https://deltadoreadb2ciot.onmicrosoft.com/iotapi/orkestrator_connector https://deltadoreadb2ciot.onmicrosoft.com/iotapi/orkestrator_endpoint https://deltadoreadb2ciot.onmicrosoft.com/iotapi/rule_management_allowed https://deltadoreadb2ciot.onmicrosoft.com/iotapi/collect_read_datas",
+                }
+            )
+
+            response = requests.post(
+                url=signin_url,
+                headers={"Content-Type": ct_header},
+                data=body,
+            )
+
+            json_response = response.json()
+            response.close()
+            access_token = json_response["access_token"]
+        
+            response = requests.get(f"https://prod.iotdeltadore.com/sitesmanagement/api/v1/sites?gateway_mac={macaddress}",
```

Comment:
Could the hard-coded `https://prod.iotdeltadore.com` root url be externalized as a constant?
Mostly to make refactoring easier if the value changes in the future.